src/libbson/src/bson/bson-timegm.c

```diff
@@ -225,47 +225,54 @@ struct rule {
 */
 
 static void
-gmtload (struct state *sp);
+gmtload (struct state *const sp);
 static struct bson_tm *
-gmtsub (const int64_t *timep, int_fast32_t offset, struct bson_tm *tmp);
+gmtsub (const int64_t *const timep,
+        const int_fast32_t offset,
+        struct bson_tm *const tmp);
 static int64_t
-increment_overflow (int64_t *number, int64_t delta);
+increment_overflow (int64_t *const ip, int64_t j);
 static int64_t
-leaps_thru_end_of (int64_t y) ATTRIBUTE_PURE;
+leaps_thru_end_of (register const int64_t y) ATTRIBUTE_PURE;
```

Comment:
I believe you are correct, but this was left unchanged to minimize changes to the warnings being addressed. I can apply a drive-by improvement to avoid the use of `restrict` if preferrable.

Reply:
Removed all use of the `register` keyword in libbson.